File: playwright-typescript/src/pages/login-page.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from '../base/base-page';
import { DashboardPage } from './dashboard-page';

export class LoginPage extends BasePage {
  private readonly usernameField: Locator;
  private readonly passwordField: Locator;
  private readonly loginButton: Locator;
  private readonly errorMessage: Locator;
  private readonly loginPageTitle: Locator;

  constructor(page: Page) {
    super(page);
    this.usernameField = page.locator('input[name="username"]');
    this.passwordField = page.locator('input[name="password"]');
    this.loginButton = page.locator('button[type="submit"]');
    this.errorMessage = page.locator('div[role="alert"]');
    this.loginPageTitle = page.locator('h5:has-text("Login")');
  }

  async enterUsername(username: string): Promise<void> {
    await this.fill(this.usernameField, username);
  }

  async enterPassword(password: string): Promise<void> {
    await this.fill(this.passwordField, password);
  }

  async clickLoginButton(): Promise<void> {
    await this.click(this.loginButton);
  }

  async login(username: string, password: string): Promise<DashboardPage> {
    await this.enterUsername(username);
    await this.enterPassword(password);
    await this.clickLoginButton();
    return new DashboardPage(this.page);
  }

  async isLoginPageDisplayed(): Promise<boolean> {
    return await this.isDisplayed(this.loginPageTitle);
  }

  async getErrorMessage(): Promise<string> {
    await this.waitForElement(this.errorMessage);
    return await this.getText(this.errorMessage);
  }

  async isErrorMessageDisplayed(): Promise<boolean> {
    return await this.isDisplayed(this.errorMessage);
  }
}

File: java-selenium/src/main/java/com/orangehrm/pages/LoginPage.java
package com.orangehrm.pages;

import com.orangehrm.base.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class LoginPage extends BasePage {

    @FindBy(name = "username")
    private WebElement usernameField;

    @FindBy(name = "password")
    private WebElement passwordField;

    @FindBy(xpath = "//button[@type='submit']")
    private WebElement loginButton;

    @FindBy(xpath = "//div[@role='alert']")
    private WebElement errorMessage;

    @FindBy(xpath = "//h5[text()='Login']")
    private WebElement loginPageTitle;

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public void enterUsername(String username) {
        sendKeys(usernameField, username);
    }

    public void enterPassword(String password) {
        sendKeys(passwordField, password);
    }

    public void clickLoginButton() {
        click(loginButton);
    }

    public DashboardPage login(String username, String password) {
        enterUsername(username);
        enterPassword(password);
        clickLoginButton();
        return new DashboardPage(driver);
    }

    public boolean isLoginPageDisplayed() {
        return isDisplayed(loginPageTitle);
    }

    public String getErrorMessage() {
        waitForElement(errorMessage);
        return getText(errorMessage);
    }

    public boolean isErrorMessageDisplayed() {
        return isDisplayed(errorMessage);
    }
}

